scripts/testing/programs/match_template/test_template_reproducibility.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python3
+"""
+Template Matching Reproducibility Test
+
+This script runs the cisTEM template matching GPU binary three times using the same input data
+and parameters, then analyzes the reproducibility by comparing the resulting MIP images.
+
+It uses the Apoferritin dataset for testing, and saves the MIP (Maximum Intensity Projection)
+files to a temporary directory with unique filenames for each replicate.
+
+The script loads the MIP images using the mrcfile package and compares them using numpy to
+measure pixel similarity between replicates with various metrics.
+"""
+
+import annoying_hack
+
+from os.path import join, exists
+from os import makedirs
+import cistem_test_utils.args as tmArgs
+import cistem_test_utils.make_tmp_runfile as mktmp
+import cistem_test_utils.run_job as runner
+import cistem_test_utils.temp_dir_manager as temp_dir_manager
+import mrcfile
+import numpy as np
+import tempfile
+import os
+import shutil
+import sys
+import re
+
+# By default the "_gpu" suffix will be added unless the --old-cistem flag is used
+# or the --cpu flag is used
+wanted_binary_name = 'match_template'
+
+
+def extract_threshold_value(hist_file_path):
+    """
+    Extract the threshold value from the histogram text file.
+
+    The threshold value is in the first line of the file, which starts with
+    "# Expected threshold = ". This function extracts the numerical value following this marker.
+
+    Args:
+        hist_file_path (str): Path to the histogram text file
+
+    Returns:
+        float: The extracted threshold value
+
+    Raises:
+        FileNotFoundError: If the file doesn't exist
+        ValueError: If the threshold value cannot be found or parsed
+    """
+    if not exists(hist_file_path):
+        raise FileNotFoundError(f"Histogram file not found at {hist_file_path}")
+
+    with open(hist_file_path, 'r') as f:
+        first_line = f.readline().strip()
+
+    # Use regex to extract the threshold value from the line
+    # Looking for a pattern like "# Expected threshold = 6.90"
+    match = re.search(r'# Expected threshold = ([\d.e+-]+)', first_line)
+    if not match:
+        raise ValueError(f"Could not find threshold value in {hist_file_path}")
+
+    threshold_value = float(match.group(1))
+    return threshold_value
+
+
+def main():
+    try:
+        # Parse command-line arguments
+        args = tmArgs.parse_TM_args(wanted_binary_name)
+
+        # Handle temp directory management options
+        if args.list_temp_dirs:
+            temp_dir_manager.print_temp_dirs()
+            return 0
+
+        if args.rm_temp_dir is not None:
```

Comment:
This block (line 79-89) seems to be generic. If it isn't already, let's make the temp_dir_manager a class and move this block into that class's constructor. We'll then need to create an instance of that class and update the calls to its methods below.